pkg/jmxfetch/jmxfetch.go

```diff
@@ -266,3 +288,30 @@ func (j *JMXFetch) Stop() error {
 func (j *JMXFetch) Wait() error {
 	return j.cmd.Wait()
 }
+
+func (j *JMXFetch) heartbeat(beat *time.Ticker) {
+	health := health.Register("jmxfetch")
+
+	for range beat.C {
+		select {
+		case <-health.C:
+		case <-j.shutdown:
+			health.Deregister()
```

Comment:
Shouldn't we call this even if we leave when `beat.C` is closed ? We could `defer`.